sql/_05_plcsql/_01_testspec/_02_declaration/_09_percent_type/cases/01_error_%type_variable.sql

```diff
@@ -0,0 +1,423 @@
+end;
+call type_support();
+
+
+EVALUATE 'ERROR-NUMERIC(38,15) : numeric value does not fit in the target type precision and scale';
+create or replace procedure type_support as
+   v_NUMERIC     type_support.T_NUMERIC%type;
+begin 
+   v_NUMERIC      := -123456789012345678901234;
+
+   dbms_output.put_line('v_NUMERIC    ' || v_NUMERIC    );
+end;
+call type_support();
+
+
+EVALUATE 'ERROR-DECIMAL : numeric value does not fit in the target type precision and scale';
+create or replace procedure type_support as
+   v_DECIMAL     type_support.T_DECIMAL%type;
+begin 
+   v_DECIMAL      := 1234567890123456;
+
+   dbms_output.put_line('v_DECIMAL    ' || v_DECIMAL    );
+end;
+call type_support();
+
+
+EVALUATE 'ERROR-DECIMAL : numeric value does not fit in the target type precision and scale';
+create or replace procedure type_support as
+   v_DECIMAL     type_support.T_DECIMAL%type;
+begin 
+   v_DECIMAL      := -1234567890123456;
+
+   dbms_output.put_line('v_DECIMAL    ' || v_DECIMAL    );
+end;
+call type_support();
+
+
+--BUG( normal : Data overflow on data type "float", BUG : -Infinity)
+EVALUATE 'ERROR : Data overflow on data type "float".';
+create or replace procedure type_support as
+   v_FLOAT       type_support.T_FLOAT%type;
+begin 
+   v_FLOAT      := -3.402823466E+39;
+
+   dbms_output.put_line('v_FLOAT    ' || v_FLOAT    );
+end;
+call type_support();
+
+
+--BUG( normal : Data overflow on data type "float", BUG : Infinity)
+EVALUATE 'ERROR : Data overflow on data type "float".';
+create or replace procedure type_support as
+   v_FLOAT       type_support.T_FLOAT%type;
+begin 
+   v_FLOAT      := 3.402823466E+39;
+
+   dbms_output.put_line('v_FLOAT    ' || v_FLOAT    );
+end;
+call type_support();
+
+
+--BUG( normal : 1.7976931348623157E+309 exceeds limit of double., BUG : -Infinity)
+EVALUATE 'ERROR : 1.7976931348623157E+309 exceeds limit of double.';
+create or replace procedure type_support as
+   v_DOUBLE      type_support.T_DOUBLE%type;
+begin 
+   v_DOUBLE      := -1.7976931348623157E+309;
+
+   dbms_output.put_line('v_DOUBLE    ' || v_DOUBLE    );
+end;
+call type_support();
+
+
+--BUG( normal : 1.7976931348623157E+309 exceeds limit of double., BUG : Infinity)
+EVALUATE 'ERROR : 1.7976931348623157E+309 exceeds limit of double.';
+create or replace procedure type_support as
+   v_DOUBLE      type_support.T_DOUBLE%type;
+begin 
+   v_DOUBLE      := 1.7976931348623157E+309;
+
+   dbms_output.put_line('v_DOUBLE    ' || v_DOUBLE    );
+end;
+call type_support();
+
+
+EVALUATE 'normal';
+create or replace procedure type_support as
+   v_DATE        type_support.T_DATE%type;
+begin 
+   v_DATE      := date'0000-00-00';
+
+   dbms_output.put_line('v_DATE    ' || v_DATE    );
+end;
+call type_support();
+drop procedure type_support;
+
+
+EVALUATE 'ERROR : invalid DATE string: 0001-01-01 00:00:00';
+create or replace procedure type_support as
+   v_DATE        type_support.T_DATE%type;
+begin 
+   v_DATE      := date'0001-01-01 00:00:00';

```

Comment:
```suggestion
   v_DATE      := '0001-01-01 00:00:00';

```
v_DATE 변수에 값 할당 단계의 유효성 체크가 아닌 '0001-01-01 00:00:00' 문자열을 date 형식으로 전환하는 과정에서 에러가 발생할 것 같습니다.
나머지 시나리오 date, time 등 날짜, 시간 타입 모두 동일

Reply:
의도한 오류입니다.
선언한 타입과 다른타입으로 처리할때의 오류를 확인하고자 하기 위함입니다.